sklearn/cluster/_hdbscan/_linkage.pyx

```diff
@@ -59,16 +59,28 @@ cpdef cnp.ndarray[MST_edge_t, ndim=1, mode='c'] mst_from_mutual_reachability(
     mst = np.empty(n_samples - 1, dtype=MST_edge_dtype)
     current_labels = np.arange(n_samples, dtype=np.int64)
     current_node = 0
+    # Contains the minimum reachability of points to the built tree. This is
+    # iteratively updated with each node we add.
     min_reachability = np.full(n_samples, fill_value=np.infty, dtype=np.float64)
     for i in range(0, n_samples - 1):
+        # Sub-select nodes not-yet in the tree
         label_filter = current_labels != current_node
         current_labels = current_labels[label_filter]
+
+        # Compute the nodes' current min-reachability scores
         left = min_reachability[label_filter]
+        # Compute the nodes' mutual-reachability to current node
         right = mutual_reachability[current_node][current_labels]
+        # Update min-reachability, given the new mutual-reachability of all
+        # nodes from the current node.
```

Comment:
Also would a better name for `left` and `right` remove the need for this comment?